### PR TITLE
Implement @methods

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is a BC break.

## Changelog

```markdown
### Changed
- Implemented `@method`s
```

### CODE REVIEW

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -2584,19 +2571,16 @@ public function getDashboardActions()
...
-     *
-     * @param string $action
-     * @param object $object
-     *
-     * @return bool
-     */
-    public function canAccessObject($action, $object)
+    public function canAccessObject(string $action, object $object): bool
     {
         return $object && $this->id($object) && $this->hasAccess($action, $object);
```
Comment (phansys) [inline src/Admin/AbstractAdmin.php]: `$object` cannot be evaluated as false anymore.
```suggestion
        return $this->id($object) && $this->hasAccess($action, $object);
```

File: src/Admin/AbstractAdmin.php [hunk 2]
```diff
@@ -1351,22 +1348,12 @@ public function getSideMenu($action, AdminInterface $childAdmin = null)
...
-     * @return string the root code
-     */
-    public function getRootCode()
+    public function getRootCode(): string
     {
         return $this->getRoot()->getCode();
     }
 
-    /**
-     * Returns the master admin.
```
Comment (greg0ire) [inline src/Admin/AbstractAdmin.php]: I think this comment should stay, it's helpful

File: src/Admin/AbstractAdmin.php [hunk 3]
```diff
@@ -2584,19 +2571,16 @@ public function getDashboardActions()
         return $filter[$name] === $default[$name];
     }
 
-    /**
-     * Check object existence and access, without throw Exception.
```
Comment (greg0ire) [inline src/Admin/AbstractAdmin.php]: Same here, the comment is helpful

File: src/Admin/AdminInterface.php [hunk 4]
```diff
@@ -725,6 +720,23 @@ public function showMosaicButton(bool $isShown): void;
      * Checks if a filter type is set to a default value.
      */
     public function isDefaultFilter(string $name): bool;
+
+    public function configureActionButtons(array $buttonList, string $action, ?object $object = null): array;
+
+    /**
+     * Check object existence and access, without throw Exception.
```
Comment (greg0ire) [inline src/Admin/AdminInterface.php]: Let's fix the grammar though
```suggestion
     * Check object existence and access, without throwing Exception.
```

File: src/Admin/AbstractAdmin.php (hunk 3 above)
Comment (core23) [reply src/Admin/AbstractAdmin.php]: It was moved tot the interface

File: src/Admin/AbstractAdmin.php (hunk 2 above)
Comment (core23) [reply src/Admin/AbstractAdmin.php]: Same here

Comment (greg0ire) [reply src/Admin/AbstractAdmin.php]: Ah my bad, it's in the interface already.

**Pull request merged**